editor/components/post-format/check.js

```diff
@@ -1,10 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { withContext } from '@wordpress/components';
+
 /**
  * Internal dependencies
  */
 import PostTypeSupportCheck from '../post-type-support-check';
 
-function PostFormatCheck( props ) {
-	return <PostTypeSupportCheck { ...props } supportKeys="post-formats" />;
+function PostFormatCheck( { themeSupportsPostFormats, ...props } ) {
+	return themeSupportsPostFormats &&
+		<PostTypeSupportCheck { ...props } supportKeys="post-formats" />;
 }
 
-export default PostFormatCheck;
+export default withContext( 'editor' )(
+	( settings ) => ( {
+		themeSupportsPostFormats: settings.themeSupportsPostFormats,
```

Comment:
Maybe destructure `settings` on L17?

Reply:
Thank you for your feedback @ockham, the improvement was applied.

Reply:
Alternatively:

```
( settings ) => pick( settings, 'themeSupportsPostFormats' ),
```

https://lodash.com/docs/4.17.5#pick

Personally, I'm on the fence about whether destructuring arguments in-place harms readability, as it obscures what it is that is being destructured (i.e. we've lost all mention of "settings").